### PR TITLE
feat(samples): Add POSIX & Manifest samples

### PR DESCRIPTION
**Pull request by danielbankhead**

### CODE REVIEW

**Review by ddelgrosso1 — APPROVED** [review]

**Review by JesseLovelace — APPROVED** [review]

File: samples/manifest-request.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  // The agent pool associated with the POSIX data source. Defaults to the default agent
+  // const sourceAgentPoolName = 'projects/my-project/agentPools/transfer_service_default'
+
+  // The root directory path on the source filesystem
+  // const rootDirectory = '/directory/to/transfer/source',
+
+  // The ID of the GCS bucket to transfer data to
+  // const gcsSinkBucket = 'my-sink-bucket'
+
+  // Transfer manifest location. Must be a
```
Comment (JesseLovelace) [inline samples/manifest-request.js:48]: Must be a what?

File: samples/manifest-request.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // Your project id
+  // const projectId = 'my-project'
+
+  // The agent pool associated with the POSIX data source. Defaults to the default agent
+  // const sourceAgentPoolName = 'projects/my-project/agentPools/transfer_service_default'
+
+  // The root directory path on the source filesystem
+  // const rootDirectory = '/directory/to/transfer/source',
```
Comment (JesseLovelace) [inline samples/manifest-request.js:43]: Trailing comma here

File: samples/posix-download.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  // const sinkAgentPoolName = 'projects/my-project/agentPools/transfer_service_default'
+
+  // Google Cloud Storage source bucket name
+  // const gcsSourceBucket = 'my-gcs-source-bucket'
+
+  // An optional path on the Google Cloud Storage bucket to download from
+  // const gcsSourcePath = 'foo/bar/'
+
+  // The root directory path on the source filesystem
+  // const rootDirectory = '/directory/to/transfer/source',
```
Comment (JesseLovelace) [inline samples/posix-download.js:49]: trailing comma

File: samples/manifest-request.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  // Transfer manifest location. Must be a
+  // const manifestLocation = 'gs://my-bucket/sample_manifest.csv'
+
+  // Creates a client
+  const client = new StorageTransferServiceClient();
+
+  /**
+   * Creates a request to transfer from the local file system to the sink bucket
+   */
+  async function transferDirectory() {
```
Comment (JesseLovelace) [inline samples/manifest-request.js:57]: Is "transferDirectory" the right name for this function? This only transfers the files specified in the manifest right? which may not be the entire directory if i understand it correctly, and may in fact include files from multiple directories within the root directory

**Pull request merged**